docs/concepts/trading-on-vega/data-sources.md

```diff
@@ -1,103 +1,113 @@
+For example, a [message taken from Coinbase's Price Oracle](https://blog.coinbase.com/introducing-the-coinbase-price-oracle-6d1ee22c7068) would have the signatures verified, and the ABI encoded data will be transformed into the following format (note: the precise representation will vary based on which API you're using):
+
+```json
+[
+	{
+		"name": "prices.BTC.timestamp",
+		"value": "1661273220"
+	},
+	{
+		"name": "prices.ETH.timestamp",
+		"value": "1661273220"
+	},
+	{
+		"name": "prices.ETH.value",
+		"value": "1635165000"
+	},
+  {
+		"name": "prices.BTC.value",
+		"value": "21420285000"
+	}
+]
+```
 
 ## Internal data source
-An internal data source provides information that comes from within Vega, rather than an external source. They are defined and used in the same way as external data sources, but are triggered by the relevant event in protocol rather than by an incoming transaction.
-
-There is currently one type of data that can come from internal data sources: time-triggered.
-
-### Internal data source: Timestamp
-Vega provides a timestamp source, which can be used to terminate a futures market at a set date. `vegaprotocol.builtin.timestamp` provides a Unix timestamp of the Vega time, which is to say the time agreed via consensus. 
-
-As the name implies, a built in data source is generated inside Vega, and cannot be submitted by other keys.
-
-This data source is used to emit an event or value, at or after the given time printed on the block.
-
-<!--### Internal data source: Value [WIP]
-This data source provides an immediate value, and is used where a data source is required, but the value is already known at the time of definition.
+Vega provides a timestamp source, which can be used to trigger a market event (such as trading termination or final settlement) at a set date and time. The `vegaprotocol.builtin.timestamp` is used by the market proposer to provide a Unix timestamp in seconds of the Vega time, which is to say the time agreed via consensus. 
+
+As the name implies, an internal data source is generated inside Vega and is submitted automatically when the time is reached.
+
+For example, a single timestamp event will appear as follows. Note: the precise representation will vary based on which API you're using):
+
+```json
+[
+	{
+		"name": "vega.builtin.timestamp",
+		"value": "1661273220"
+	}
+]
+```
+
+## Signed JSON message
+Signed JSON messages are an alternative to using *signed message data sources*, which are best used for off-chain prices. Some markets need data other than prices, and for that flexibility Vega also supports arbitrary JSON messages, signed by a Vega key. Data from these types is only as reliable as the Vega public key that publishes the data, whereas signed message data sources have signed data. The advantage is that the format is less rigid than signed messages: any properties can be specified, including strings, booleans or non-price numbers.
```

Comment:
Signed JSON messages can and should also be used for prices if they don't come from an Open Oracle source, as they are easier to work with. So anyone creating an oracle afresh just for Vega would probably use these for their price source.